pkg/kubediscovery/handlers.go

```diff
@@ -63,75 +41,137 @@ type caLoader interface {
 // fsCALoader is a caLoader for loading the PEM encoded CA from
 // /tmp/secret/ca.pem.
 type fsCALoader struct {
+	certData string
 }
 
 func (cl *fsCALoader) LoadPEM() (string, error) {
-	file, err := os.Open(CAPath)
-	if err != nil {
-		return "", err
+	if cl.certData != "" {
+		data, err := ioutil.ReadFile(CAPath)
+		if err != nil {
+			return "", err
+		}
+
+		cl.certData = base64.StdEncoding.EncodeToString(data)
```

Comment:
I think I see why you made the change in Joe's proposal, but there may be an issue there, I've asked for opinions here: https://github.com/kubernetes/kubernetes/pull/30707/files#diff-b8a5644fded1432a29870c53b5644190R57